node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -345,17 +355,30 @@ fn configure_database(
 }
 
 fn configure_cryptdes(
+    new_public_key: Option<bool>,
     persistent_config: &mut dyn PersistentConfiguration,
     db_password_opt: &Option<String>,
 ) -> CryptDEPair {
     let cryptde_pair = if let Some(db_password) = db_password_opt {
         let chain = Chain::from(persistent_config.chain_name().as_str());
-        let main_result = persistent_config.cryptde(db_password);
+        let main_result = match new_public_key {
+            None | Some(false) => persistent_config.cryptde(db_password),
+            Some(true) => {
```

Comment:
User wants a new public key. So we don't care what's currently in the database; we just make a new `CryptDEReal`, store it in the database over whatever was there before, and return it from the `match`.